### DISCUSSION TITLE
Model regresi tidak memenuhi uji asumsi, tapi jika beberapa prediktor dihilangkan, nilai adj R squared jadi rendah

### DISCUSSION THREAD

**Discussion by galuhchynintya** [teamalgoritma/community#889] [Q&A] [answered]
Dear tim algoritma,
Saya sedang mengerjakan LBB RM, tapi ternyata model yang saya buat tidak memenuhi uji asumsi, jika saya menghilangkan beberapa prediktor (ambil prediktor yg punya nilai VIF kurang dari 10 saja), setelah dibuat model, nilai adj R squared jadi rendah. Apa yang sebaiknya saya lakukan?

**Comment by IrfanChairurrachman** [comment]
Dear @galuhchynintya ,

Jika model linear regression masih belum memenuhi asumsi no multiconlinearity, ada beberapa hal yang dapat dilakukan:

1. Membuat model kembali **menggunakan salah satu prediktor** yang terindikasi multiko. Sehingga jika terdapat 2 prediktor yg nilai VIF nya lebih dari 10, Anda dapat menggunakan salah satunya saja.
2. Tidak menggunakan prediktor-prediktor tersebut. Mengambil semua prediktor yang nilai VIFnya kurang dari 10, seperti yang Anda lakukan)
3. Melakukan feature engineering dengan membuat variabel baru yang berisi rata-rata dari kedua variabel yang terindikasi multiko. Contoh: Prediktor D dan E terindikasi multiko, Anda dapat membuat kolom Z yg berisi nilai rata-rata (atau cara lain) dari D & E. Kemudian menggunakan Z menjadi prediktor (dan take out D & E sebagai prediktor)
4. Gunakan model lain yang lebih kompleks (model yang bebas asumsi). Contoh: seperti model-model Decision Tree Regressor atau model lainnya. Hanya saja ini Anda dapat lakukan ketika ke-3 cara di atas masih belum memberikan hasil yang baik.

Semoga informasi ini dapat membantu. Bila ternyata informasi ini dapat menjawab pertanyaan Anda, Anda dapat “Mark as answer” jawaban ini.

**Comment by galuhchynintya** [comment]
Untuk uji asumsi, apakah semuanya harus terpenuhi mas Irfan?

On Mon, Jul 22, 2024 at 11:26 AM Irfan Chairur Rachman <
***@***.***> wrote:

> Dear @galuhchynintya <https://github.com/galuhchynintya> ,
>
> Jika model linear regression masih belum memenuhi asumsi no
> multiconlinearity, ada beberapa hal yang dapat dilakukan:
>
>    1. Membuat model kembali *menggunakan salah satu prediktor* yang
>    terindikasi multiko. Sehingga jika terdapat 2 prediktor yg nilai VIF nya
>    lebih dari 10, Anda dapat menggunakan salah satunya saja.
>    2. Tidak menggunakan prediktor-prediktor tersebut. Mengambil semua
>    prediktor yang nilai VIFnya kurang dari 10, seperti yang Anda lakukan)
>    3. Melakukan feature engineering dengan membuat variabel baru yang
>    berisi rata-rata dari kedua variabel yang terindikasi multiko. Contoh:
>    Prediktor D dan E terindikasi multiko, Anda dapat membuat kolom Z yg berisi
>    nilai rata-rata (atau cara lain) dari D & E. Kemudian menggunakan Z menjadi
>    prediktor (dan take out D & E sebagai prediktor)
>    4. Gunakan model lain yang lebih kompleks (model yang bebas asumsi).
>    Contoh: seperti model-model Decision Tree Regressor atau model lainnya.
>    Hanya saja ini Anda dapat lakukan ketika ke-3 cara di atas masih belum
>    memberikan hasil yang baik.
>
> Semoga informasi ini dapat membantu. Bila ternyata informasi ini dapat
> menjawab pertanyaan Anda, Anda dapat “Mark as answer” jawaban ini.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/teamalgoritma/community/discussions/889#discussioncomment-10110446>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BHP3DD6C35YK2L2DYZBJIXTZNSCYZAVCNFSM6AAAAABLF2LYVSVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTAMJRGA2DINQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***
> com>
>


-- 
Sincerely,

Galuh Chynintya R.P.

**Reply by IrfanChairurrachman** [reply, marked as answer]
@galuhchynintya ,

Untuk uji asumsi semuanya tidak harus terpenuhi kok Bu. Minimal uji asumsi yang harus terpenuhi agar model dapat digunakan adalah:

- Linearity
- No Multinolinearity

Namun, untuk LBB, tidak ada masalah jika model tidak memenuhi uji asumsi, yang terpenting Anda dapat membuat projek, menjelaskan dan menarik kesimpulan dari setiap tahapan-tahapan yang Anda lakukan.

**Reply by galuhchynintya** [reply]
Berikut beberapa model yang sudah di buat
1. model_backward
![image](https://github.com/user-attachments/assets/359ff073-e08f-4344-92ad-79d8f8f10cb2)

linearity lulus, normality using saphiro test (p-value < alpha (0.05)), Homoscedasticity using bp test (p-value < alpha (0.05))

Menggunakan salah satu prediktor yang nilai VIF nya lebih dari 10 (Volume berkolerasi dengan log_volume, open berkorelasi dengan priorday_adjclose) 
2. model_trial 1 (prediktor: range, log_volume, return_percentage, open, priorday_adjclose)
![image](https://github.com/user-attachments/assets/10243343-9b32-4ecf-a93b-c032f200ed16)

adj r square: 0.999
linearity lulus, normality using saphiro test (p-value < alpha (0.05)), Homoscedasticity using bp test (p-value < alpha (0.05))

3. model_trial1(prediktor: range, log_volume, return_percentage, priorday_adjclose)
![image](https://github.com/user-attachments/assets/4114cf9b-58f5-4fd2-94a0-0721f014519f)

adj r square: 0.999
linearity lulus, normality using saphiro test (p-value < alpha (0.05)), Homoscedasticity using bp test (p-value < alpha (0.05))

4. model_trial1(prediktor: range, log_volume, return_percentage, open)
![image](https://github.com/user-attachments/assets/4f7e2070-a00a-4d7b-8292-08d01c4cc45d)

adj r square: 0.9981
linearity lulus, normality using saphiro test (p-value < alpha (0.05)), Homoscedasticity using bp test (p-value < alpha (0.05))

Kalau begitu dengan melihat nilai VIF, saya bisa gunakan model no 3 atau 4 ya, tapi dengan melihat nilai adj r square, lebih baik model no 3.
Terima kasih penjelasannya mas Irfan
